File: docs/data-structures/arrays/code/cpp/array_examples.cpp
#include <array>
#include <iostream>
#include <optional>

std::optional<std::size_t> linear_search(const std::array<int, 5>& values, int target) {
    for (std::size_t index = 0; index < values.size(); ++index) {
        if (values[index] == target) {
            return index;
        }
    }
    return std::nullopt;
}

int main() {
    std::array<int, 5> values{12, 27, 31, 45, 58};

    std::cout << "Traversal: ";
    for (int value : values) {
        std::cout << value << ' ';
    }
    std::cout << '\n';

    const int target = 31;
    const std::optional<std::size_t> result = linear_search(values, target);

    if (result.has_value()) {
        std::cout << "Found " << target << " at index " << result.value() << '\n';
    } else {
        std::cout << target << " was not found\n";
    }

    return 0;
}

File: docs/data-structures/arrays/code/python/array_examples.py
def linear_search(values: list[int], target: int) -> int:
    for index, value in enumerate(values):
        if value == target:
            return index
    return -1


def main() -> None:
    values = [12, 27, 31, 45, 58]

    print("Traversal:", end=" ")
    for value in values:
        print(value, end=" ")
    print()

    target = 31
    index = linear_search(values, target)

    if index != -1:
        print(f"Found {target} at index {index}")
    else:
        print(f"{target} was not found")

    values.insert(2, 25)
    print("After insertion:", values)

    del values[2]
    print("After deletion:", values)


if __name__ == "__main__":
    main()